## Sampling Error

Sampling error is the difference in statistics (sample information) and the actually population parameters.

`sampling error = parameter - statistic`

### Task One

Given the data wnba.csv in the folder data.

The dataset is about basketball players in the WNBA (Women's National Basketball Association), and contains general information about players, along with their metrics for the 2016-2017 season.

Explore the dataset.

+ Print the first five rows using DataFrame.head() and the last five rows with DataFrame.tail().
+ Find the number of rows and columns using DataFrame.shape.
+ Learn about each column from the documentation. You can also find useful documentation in this glossary and on WNBA's official page.

Take one measure of the sampling error.

+ Use the Games Played column to find the maximum number of games played by a player in the 2016-2017 season. The dataset contains all the players who played at least one game, so it's a population relative to our question. Find this parameter, and assign the result to a variable named parameter.
+ Using the `DataFrame.sample()` method, randomly sample 30 players from the population, and assign the result to a variable named sample.
+ When calling `Series.sample()`, use the argument random_state = 1. This makes your results reproducible, and it helps us with the answer-checking (we'll discuss this on the next screen).
+ Find the maximum number of games using the sample, and assign the result to a variable named statistic.
+ Measure the sampling error, and assign the result to a variable named sampling_error.


In [2]:
import pandas as pd

path = "data/wnba.csv"

wnba = pd.read_csv(path)

In [3]:
wnba.head()

Name Team  Pos  Height  Weight        BMI Birth_Place  \
0    Aerial Powers  DAL    F     183    71.0  21.200991          US   
1      Alana Beard   LA  G/F     185    73.0  21.329438          US   
2     Alex Bentley  CON    G     170    69.0  23.875433          US   
3  Alex Montgomery  SAN  G/F     185    84.0  24.543462          US   
4     Alexis Jones  MIN    G     175    78.0  25.469388          US   

           Birthdate  Age         College  ... OREB  DREB  REB  AST  STL  BLK  \
0   January 17, 1994   23  Michigan State  ...    6    22   28   12    3    6   
1       May 14, 1982   35            Duke  ...   19    82  101   72   63   13   
2   October 27, 1990   26      Penn State  ...    4    36   40   78   22    3   
3  December 11, 1988   28    Georgia Tech  ...   35   134  169   65   20   10   
4     August 5, 1994   23          Baylor  ...    3     9   12   12    7    0   

   TO  PTS  DD2  TD3  
0  12   93    0    0  
1  40  217    0    0  
2  24  218    0    0  
3  38  188    2    0  
4  14   50    0    0  

[5 rows x 32 columns]

In [4]:
wnba.tail()

Name Team  Pos  Height  Weight        BMI Birth_Place  \
138     Tiffany Hayes  ATL    G     178    70.0  22.093170          US   
139   Tiffany Jackson   LA    F     191    84.0  23.025685          US   
140  Tiffany Mitchell  IND    G     175    69.0  22.530612          US   
141      Tina Charles   NY  F/C     193    84.0  22.550941          US   
142     Yvonne Turner  PHO    G     175    59.0  19.265306          US   

              Birthdate  Age         College  ... OREB  DREB  REB  AST  STL  \
138  September 20, 1989   27     Connecticut  ...   28    89  117   69   37   
139      April 26, 1985   32           Texas  ...    5    18   23    3    1   
140  September 23, 1984   32  South Carolina  ...   16    70   86   39   31   
141        May 12, 1988   29     Connecticut  ...   56   212  268   75   21   
142    October 13, 1987   29        Nebraska  ...   11    13   24   30   18   

     BLK  TO  PTS  DD2  TD3  
138    8  50  467    0    0  
139    3   8   28    0    0  
140    5  40  277    0    0  
141   22  71  582   11    0  
142    1  32  151    0    0  

[5 rows x 32 columns]

In [5]:
wnba.shape

(143, 32)

the dataset has 143 rows and 32 columns

Maximum number of games played in the whole dataset

In [6]:
parameter = wnba['Games Played'].max()

Generating a random sample of games player for 30 players

In [7]:
sample = wnba['Games Played'].sample(30, random_state=1)

Sample maximum number of games played

In [8]:
statistic = sample.max()

The sampling error is the difference between the two

In [10]:
sampling_error = parameter - statistic

sampling_error

2

## Random Sampling